src/stream/src/executor/managed_state/aggregation/extreme.rs

```diff
@@ -154,13 +157,17 @@ impl<S: StateStore> GenericExtremeState<S> {
         }
     }
 
-    fn state_row_to_cache_entry(&self, state_row: &Row) -> (OrderedRow, Datum) {
-        let cache_key = OrderedRow::new(
-            state_row.by_indices(&self.state_table_order_col_indices),
-            &self.state_table_order_types,
+    fn state_row_to_cache_entry(&self, state_row: &Row) -> StreamExecutorResult<(Vec<u8>, Datum)> {
+        let mut cache_key = Vec::new();
+        let serializer = OrderedRowSerializer::new(self.state_table_order_types.clone());
```

Comment:
That'll be costly as we clone the order types every time this function is called. Looks like the `state_table_order_types` is only used here, I think it's better to simply store an instance of `OrderedRowSerializer` in self.